### DISCUSSION TITLE
FR: Improve ease of onboarding by splitting files and moving deployment_name to the top of the file

### DISCUSSION THREAD

**Discussion by sweiss-google** [GoogleCloudPlatform/cluster-toolkit#3135] [Ideas and feature requests]
A customer using cluster toolkit for the first time had a few requests that they say would have improved their onboarding journey.

X-posted with their approval:

> Concrete recommendations:
>
> 1. Split the deployment-image-cluster.yaml file into two files: deployment-image.yaml and deployment-cluster.yaml
> 2. In each of those files, make the vars section start with deployment_name, e.g. deployment_name: a3mega-image and deployment_name: a3mega-cluster as it does in the deployment-base.yaml.
> 3. In the GCP Documentation, make deployment_name, final_image_family, slurm_cluster_name specifiable by the user when interpolating the deployment files.

**Comment by tpdownes** [comment]
My intent is to combine the 3 blueprints into 1 once we are able to adopt Filestore deletion protection in the Google terraform plugin v6.4.0:

https://github.com/hashicorp/terraform-provider-google/releases/tag/v6.4.0

No precise timeline, but there is an open PR #3107 to make our code base plugin v6 compatible. Should be short work to add the flag to our Filestore module.

I imagine this (1 blueprint) would be preferable to the suggestion.

FYI: the reason for having "2 deployment files in 1" is that they would otherwise contain duplicate information that **must** match.
